src/storage/src/hummock/compactor/fast_compactor_runner.rs

```diff
@@ -466,13 +466,8 @@ impl CompactorRunner {
                 let smallest_key = FullKey::decode(sstable_iter.next_block_smallest()).to_vec();
                 let (block, filter_data, block_meta) =
                     sstable_iter.download_next_block().await?.unwrap();
-                if self.executor.builder.need_flush() {
-                    let largest_key = sstable_iter.sstable.value().meta.largest_key.clone();
-                    let target_key = FullKey::decode(&largest_key);
-                    sstable_iter.init_block_iter(block, block_meta.uncompressed_size as usize)?;
-                    let mut iter = sstable_iter.iter.take().unwrap();
-                    self.executor.run(&mut iter, target_key).await?;
-                } else {
+                let is_new_user_key = !self.executor.last_key.user_key.eq(&smallest_key.user_key);
+                if !self.executor.builder.need_flush() && is_new_user_key {
```

Comment:
Could you please add some comments here? It's kind of confusing.

Reply:
Done. I have add comment for it.